atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/ValidatingQueryRewritingKeyValueService.java

```diff
@@ -97,10 +102,11 @@ public void createTables(Map<TableReference, byte[]> tableRefToTableMetadata) {
     protected void sanityCheckTableName(TableReference tableRef) {
         String tableName = tableRef.getQualifiedName();
         Validate.isTrue(
-                (!tableName.startsWith("_") && tableName.contains("."))
+                (!tableName.startsWith("_") && (tableName.contains(".")))
+                        && ALLOWED_TABLE_NAMES.matcher(tableName).matches()
                         || AtlasDbConstants.hiddenTables.contains(tableRef)
                         || tableName.startsWith(AtlasDbConstants.NAMESPACE_PREFIX),
-                "invalid tableName: %s", tableName);
```

Comment:
this was just wrong before, Validate doesn't take format strings.
Not sure if this should also be fixed for safe logging, though